### PR TITLE
add jojo tvl

### PR DESCRIPTION
**Pull request by JoscelynFarr**
**NOTE**

#### Please enable "Allow edits by maintainers" while putting up the PR.

---

> - If you would like to add a `volume` adapter please submit the PR [here](https://github.com/DefiLlama/adapters).
> - If you would like to add a `liquidations` adapter, please refer to [this readme document](https://github.com/DefiLlama/DefiLlama-Adapters/tree/main/liquidations) for details.

1. Once your adapter has been merged, it takes time to show on the UI. If more than 24 hours have passed, please let us know in Discord.
2. Sorry, We no longer accept fetch adapter for new projects, we prefer the tvl to computed from blockchain data, if you have trouble with creating a the adapter, please hop onto our discord, we are happy to assist you.
3. Please fill the form below  **only if the PR is for listing a new protocol** else it can be ignored/replaced with reason/details about the PR
4. **For updating listing info** It is a different repo, you can find your listing in this file: https://github.com/DefiLlama/defillama-server/blob/master/defi/src/protocols/data2.ts, you can  edit it there and put up a PR
5. Do not edit/push `package-lock.json` file as part of your changes, we use lockfileVersion 2, and most use v1 and using that messes up our CI
6. No need to go to our discord and announce that you've created a PR, we monitor all PRs and will review it asap

---
##### Logo (High resolution, will be shown with rounded borders): 

green_square_icon

##### Short Description (to be shown on DefiLlama):
JOJO is a decentralized perpetual contract trading platform that uses an off-chain matching and on-chain settlement order book model. JOJO balances liquidity and composability, providing professional traders with a smooth, fast, and low-cost trading experience and hedging capabilities for other DeFi protocols. JOJO is where it's at for serious traders.

##### Implementation Details: Briefly describe how the oracle is integrated into your project: 

Basically, JOJO uses Chainlink as the primary decentralized oracle provider and Uniswap V3 as a complement, with JOJO self-built oracle to serve as an emergency oracle.
##### Documentation/Proof: Provide links to documentation or any other resources that verify the oracle's usage: 
https://about.jojo.exchange/jojo-1/trade/pricing

### CODE REVIEW

**Comment by llamatester** [comment]
Error while running adapter at projects/jojo: 
        
 
  
    
    
    TypeError: Cannot read properties of undefined (reading 'toLowerCase')
    Truncated error stack:
    
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:121:50
        at Array.forEach (<anonymous>)
        at getUniqueAddresses (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:121:13)
        at sumTokens2 (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:708:12)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:811:41
        at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:64:29)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:160:15
        at Array.map (<anonymous>)
        at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:148:29
        at Array.map (<anonymous>)

**Comment by JoscelynFarr** [comment]
> Error while running adapter at projects/jojo:
> 
> ```
> TypeError: Cannot read properties of undefined (reading 'toLowerCase')
> Truncated error stack:
> 
>     at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:121:50
>     at Array.forEach (<anonymous>)
>     at getUniqueAddresses (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/tokenMapping.js:121:13)
>     at sumTokens2 (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:708:12)
>     at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/projects/helper/unwrapLPs.js:811:41
>     at getTvl (/home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:64:29)
>     at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:160:15
>     at Array.map (<anonymous>)
>     at /home/runner/work/DefiLlama-Adapters/DefiLlama-Adapters/test.js:148:29
>     at Array.map (<anonymous>)
> ```

Hey I did not change the code in here, is there any way to fix it?

**Comment by llamatester** [comment]
The adapter at projects/jojo exports TVL: 
        
 
 
    era                       0
    bsc                       0.0017390074739344775
    arbitrum                  842.67 k
    
    total                    842.67 k

**Comment by JoscelynFarr** [comment]
Yes that's right!

**Comment by JoscelynFarr** [comment]
Hey @llamatester is that ok to update the TVL of JOJO, thanks!

**Comment by g1nt0ki** [comment]
@JoscelynFarr thanks for the PR
